### PR TITLE
Example of Monte Carlo sampling using RO ERD flowsheet

### PR DESCRIPTION
**Pull request by eyoung55**
## Fixes/Addresses:

Adds a script demonstrating how to perform Monte Carlo sampling of an existing flowsheet using RO with energy recovery as an example.

## Summary/Motivation:

It's useful to see a working example of how to use the random sample classes and how to interface with an actual flowsheet.

## Changes proposed in this PR:
- Add an example Monte Carlo analysis of an existing flowsheet.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by adam-a-a** [comment]
This looks good to me and could probably be transformed directly into a how-to for our documentation (with some minor additions). What do you think, @eyoung55 @bknueven @jefalon ?

File: proteuslib/flowsheets/RO_with_energy_recovery/monte_carlo_sampling_RO_ERD.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+initialize_system(m, solver=solver)
+
+# Simulate once outside the parameter sweep to ensure everything is appropriately initialized 
+solve(m, solver=solver)
+
+# Define the sampling type and ranges for three different variables
+sweep_params = {}
+sweep_params['A_comp'] = NormalSample(m.fs.RO.A_comp, 4.0e-12, 0.5e-12)
+sweep_params['B_comp'] = NormalSample(m.fs.RO.B_comp, 3.5e-8, 0.5e-8)
+sweep_params['Porosity'] = UniformSample(m.fs.RO.spacer_porosity, 0.95, 0.99)
```
Comment (adam-a-a) [inline proteuslib/flowsheets/RO_with_energy_recovery/monte_carlo_sampling_RO_ERD.py:25]:
```suggestion
sweep_params['Spacer_porosity'] = UniformSample(m.fs.RO.spacer_porosity, 0.95, 0.99)
```
I know it's just an alias but would want users with desal expertise to have no confusion about which porosity we are talking about (even though one could easily observe `spacer_porosity` later in the line).

Comment (eyoung55) [reply proteuslib/flowsheets/RO_with_energy_recovery/monte_carlo_sampling_RO_ERD.py]: Makes sense to me! Fixed.

**Comment by bknueven** [comment]
@adam-a-a I think I agree -- perhaps this file broken out could replace what's there, which is already somewhat broken anyways (see #133 and #161).

**Comment by adam-a-a** [comment]
@bknueven That could work. Just to clarify, are you referring to replacing the how-to for parameter sweep with a slightly beefed-up version of this?  

I think it wouldn't hurt to have two separate how-to docs: one that introduces the use of parameter sweep (w/LinearSample), and a second follow-on how-to that shows how to perform Monte Carlo sampling via this example. On the one hand, it could seem a little repetitive, but on the other hand, I think it would really drive the teaching points home for the user.

**Comment by bknueven** [comment]
@adam-a-a Given that `LinearSample` and the random sampling types cannot be combined, I think you are correct that there should be two separate how-to's:
1. How to systematically sample flowsheet parameters
2. How to Monte Carlo sample flowsheet parameters.

**Comment by adam-a-a** [comment]
@bknueven That could work. Just to clarify, are you referring to replacing the how-to for parameter sweep with a slightly beefed-up version of this?  

I think it wouldn't hurt to have two separate how-to docs: one that introduces the use of parameter sweep, and a second follow-on how-to that shows how to perform Monte Carlo sampling 

> @adam-a-a Given that `LinearSample` and the random sampling types cannot be combined, I think you are correct that there should be two separate how-to's:
> 
> 1. How to systematically sample flowsheet parameters
> 2. How to Monte Carlo sample flowsheet parameters.

@bknueven Sounds good to me.

**Comment by bknueven** [comment]
The comments about documentation aside, I think there is value in having this example. @eyoung55 could you add a test for this in `proteuslib/flowsheets/RO_with_energy_recovery/tests` just so this code doesn't accidentally go stale?

**Comment by eyoung55** [comment]
Just getting up to speed on Friday's conversation.  @bknueven, yes I can add a test with this code.  Is the idea that this script should effectively live as a test only, with some additional explanation being added to the existing how-to guide on random sampling?  Or is there still value in having this as a standalone script `proteuslib/flowsheets/RO_with_energy_recovery/monte_carlo_sampling_RO_ERD.py `?  I'm thinking the former, but let me know if this idea has changed.

**Comment by bknueven** [comment]
I think there is some value in having this whole Python modules as example. This one is so small the entire code block could just be displayed verbatim in a how-to guide, and we could also give an example of running it in parallel.

**Review by adam-a-a — APPROVED** [review]
LGTM. As the purpose of this PR was to show an example only, my previous points on directly building how-to docs from this can be resolved in a subsequent PR. 
Perhaps we should create an issue for creating those how-to docs in case one doesn't exist already.

**Review by bknueven — APPROVED** [review]

**Pull request merged**